osutil/io_test.go

```diff
@@ -246,41 +246,58 @@ func (ts *AtomicSymlinkTestSuite) TestAtomicSymlink(c *C) {
 		c.Check(exp, Equals, target)
 	}
 
+	checkLeftoverFiles := func(sym string, exp []string) {
+		res, err := filepath.Glob(sym + "*")
```

Comment:
I think the function could take advantage of the fact our temporary files have ~ in their name and just fail in case _any_ such file is present.

Reply:
Well, but the temporary files/symlinks all have "~" and some random string appended to the end of the filepath they are creating, so if we did look only for files with "~", we could end up failing if `AtomicSymlink` no longer uses "~" in their filename. Doing it this way I think is more robust IMHO.